CRM/Core/BAO/CustomField.php

```diff
@@ -1200,7 +1200,7 @@ private static function formatDisplayValue($value, $field, $entityId = NULL) {
     }
     // CRM-12989 fix
     if ($field['html_type'] == 'CheckBox') {
-      CRM_Utils_Array::formatArrayKeys($value);
+      CRM_Utils_Array::convertCheckboxFormatToArray($value);
```

Comment:
@mlutfy ..even I had to spent some time to figure out the test failures, and then I realised its a return function, so you need to change it to:
```
$value = CRM_Utils_Array::convertCheckboxFormatToArray($value);
```

On the other hand, I found that for this case, the values are already sent on proper format and this fn. simply returns it but still its better to have ```convertCheckboxFormatToArray()``` in check.

Reply:
ping @mlutfy